tools/print_test_stats.py

```diff
@@ -662,10 +661,23 @@ def parse_report(path: str) -> Iterator[TestCase]:
         yield TestCase(test_case)
 
 
+def get_recursive_files(folder: str, extension: str) -> Iterable[str]:
+    """
+    Get recursive list of files with given extension even.
+
+    Use it instead of glob(os.path.join(folder, '**', f'*{extension}'))
+    if folder/file names can start with `.`, which makes it hidden on Unix platforms
```

Comment:
this is because some of the test files defined in test/run_test.py has folder structure, such as `distributed/test_distributed_spawn.py` is named as `distributed.test_distributed_spawn`. 

In order to deal with this #52791 introduce a filename sanitizer to replace all `'/'` with `'.'`, this works fine until `inspect.getfile` no longer returns relative path. It should be fixed by #60380. 

but I think having this here will still capture wrongly generated test filename in the future.

Reply:
`distributed.test_distributed_spawn` should be fine, but `.distributed.test_distributed_spawn/` will be skipped by `glob`

Reply:
it won't be `.distributed.test_distributed_spawn/`. 
to answer @janeyx99 's question as well: actually the one that starts with `.` was: `.var.lib.jenkins.workspace.test.distributed.test_distributed_spawn` because it was originally `/var/lib/jenkins/workspace/test/distributed/test_distributed_spawn` and with the regex replace all `/` became `.`